code/src/hardware_usart.cpp

```diff
@@ -9,18 +9,17 @@ namespace r2d2 {
 
     void hardware_usart_c::send_byte(const uint8_t &b) {
         while (!transmit_ready()) {
-            hardware_usart->US_THR = b;
         }
+        hardware_usart->US_THR = b;
     }
 
     uint8_t hardware_usart_c::receive_byte() {
         return hardware_usart->US_RHR;
     }
 
-    hardware_usart_c::hardware_usart_c(unsigned int baudrate, uart_ports_c &usart_port):
-        baudrate(baudrate),
-        usart_port(usart_port)
-    {
+    hardware_usart_c::hardware_usart_c(unsigned int baudrate,
```

Comment:
this constructor looks weird.. please repair this. sorry for not noticing earlier

Reply:
something for the style guide?
or should this be updated in clang?
will change it anyway